src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -165,5 +172,634 @@ ConvSolution InitAnyInvokerFactory(const ProblemDescriptionType& problem,
+                                  const miopen::conv::ProblemDescription& problem,
+                                  const CKArgsType& ck_args,
+                                  const TransposeOps&... transpose_ops)
+{
+    return MakeTaggedTransposeInstancesHelper<CKArgsType>(
+        result,
+        ctx,
+        problem,
+        ck_args,
+        std::make_tuple(transpose_ops...),
+        std::index_sequence_for<TransposeOps...>{});
+}
+*/
+
+template <typename CKArgsType,
+          typename Input1TposeOp,
+          typename Input2TposeOp,
+          typename OutputTposeOp>
+auto MakeTaggedTransposeInstances(ConvSolution& result,
+                                  const ExecutionContext& ctx,
+                                  const miopen::conv::ProblemDescription& problem,
+                                  const CKArgsType& ck_args,
+                                  const Input1TposeOp& input1_op,
+                                  const Input2TposeOp& input2_op,
+                                  const OutputTposeOp& output_op)
+{
+
+    auto input1_solver = input1_op.MakeTransposeSolver(ctx, problem, ck_args);
+    auto input2_solver = input2_op.MakeTransposeSolver(ctx, problem, ck_args);
+    auto output_solver = output_op.MakeTransposeSolver(ctx, problem, ck_args);
+
+    // NOTE: In cases where the convolution updates only a subset of output
+    // indices, we need to first initialize the workspace buffer for
+    // output with the real tensor for the output and then apply the convolution.
+    // This is achieved by creating an input transpose op for the output workspace
+    // bufffer.
+
+    using OutputInitOp = CKTransposeInputOp<OutputTposeOp::NDIM, OutputTposeOp::CONV_OP_TAG>;
+
+    auto output_init_solver = OutputInitOp{}.MakeTransposeSolver(ctx, problem, ck_args);
+
+    result.construction_params.insert(result.construction_params.end(),
+                                      {input1_solver.GetKernelInfo(),
+                                       input2_solver.GetKernelInfo(),
+                                       output_solver.GetKernelInfo(),
+                                       output_init_solver.GetKernelInfo()});
+
+    constexpr size_t buf_alignment = 256ull;
+    MultiBufferWorkspaceTraits wt({input1_solver.GetOutputTensorSize(),
+                                   input2_solver.GetOutputTensorSize(),
+                                   output_solver.GetOutputTensorSize()},
+                                  buf_alignment);
+
+    return std::make_tuple(
+        TransposeInstanceTagged{input1_solver, 0, wt, 0, Input1TposeOp::CONV_OP_TAG},
+        TransposeInstanceTagged{input2_solver, 1, wt, 1, Input2TposeOp::CONV_OP_TAG},
+        TransposeInstanceTagged{output_solver, 2, wt, 2, OutputTposeOp::CONV_OP_TAG},
+        TransposeInstanceTagged{output_init_solver, 3, wt, 2, OutputTposeOp::CONV_OP_TAG});
+}
+
+template <typename V>
+void DebugPrintVec(const char* name, const V& vec)
```

Comment:
[Recommendation] Enclose the whole debugging-only block of code into
```C++
#if 0 /// This code should not be used in the production.
...
#endif
```
Why: It can be easy for developers to forget to remove calls to `DebugPrintCKArgPtrs()` or `DebugPrintConvTensors()` in the future.

Reply:
I've wrapped it in `#ifndef NDEBUG` so that the code still compiles in debug mode but is disabled for release builds.

Reply:
[resolved]